server/routes/notification.js

```diff
@@ -0,0 +1,224 @@
+    //inside the hashmap is equal to number of users who has the particular product in their
+    //list regardless of how many list it is in.
+    //We use a hashmap to provide better look up times since we'll have to update data as we go
+    let newNotifs = {};
+
+    //This is an array of all owners of each of those list. Non duplicating lists but with a
+    //possibility of users owning more than one of those list.
+    const userLists = await UserListModel.findAll({
+      where: {
+        id: listProds.map((listProd) => {
+          return listProd.list_id;
+        }),
+      },
+      transaction: trans,
+    });
+
+    //For each of the over all users in the userLists, we build the data object which will be
+    //placed inside the NotificationModel and its other attributes. This iterates for each of the userLists, but conditionals
+    //are placed to only make one per user. If the model a schema has already been made for a user, we instead
+    //make an update to the objects data.listLocations
+    userLists.forEach((userList) => {
+      //If we havent made a data object for the user, make one.
+      if (!newNotifs[`${userList.user_id}`]) {
+        const data = {
+          title,
+          productId,
+          price,
+          previousPrice,
+          listLocations: [userList.id],
+        };
+
+        newNotifs[`${userList.user_id}`] = {
+          type: "price",
+          data,
+          user_id: userList.user_id,
+          isRead: false,
+        };
+
+        //The data object has an array listLocations. This is basically the lists of a user where the
+        //particular product is placed. If a data object is already made, we simply add in the current list
+        //to the listLocations array.
+      } else {
+        newNotifs[`${userList.user_id}`].data.listLocations.push(userList.id);
+      }
+    });
+
+    //This would convert the newNotifs object into an array.
+    //[ [dataObject], [dataObject], ... ]
+    //Inside the data object, the [0] index is the key, while the [0] index is the data object itself.
+    const parsedNotifs = Object.entries(newNotifs);
+
+    const notifPromiseArr = [];
+
+    //This iteration is the main part of creating the notification
+    parsedNotifs.forEach((notif) => {
+      notifPromiseArr.push(
+        NotificationModel.create(notif[1], { transaction: trans })
+      );
+    });
+
+    //This is the array of all the new notifications made for a user.
+    //May be changed in the future to just [1] for success since no one need to know
+    const allNewNotifs = await Promise.all(notifPromiseArr);
+
+    //Update Product Model
+    await ProductModel.update(
+      {
+        current_price: price,
+        previous_price: productModel.current_price,
+      },
+      {
+        where: {
+          id: productId,
+        },
+        transaction: trans,
+      }
+    );
+
+    await trans.commit();
+
+    res.status(201).send(allNewNotifs);
```

Comment:
in this case I don't think we want to send all the notifications back. You can imagine that at some point this could be an array of thousands of notification objects and that's not necessary or efficient

Reply:
Replaced sent data to an object.

`
{msg: "Success"} 
`